tests/conftest.py

```diff
@@ -17,6 +18,128 @@
 os.environ["VLLM_WORKER_MULTIPROC_METHOD"] = "spawn"
 
 
+def pytest_generate_tests(metafunc):
+    """This hook is called during the collection phase, 
+    specifically when Pytest encounters a test function that 
+    needs parametrization. It receives a metafunc object, 
+    which provides information about the test function and 
+    allows for dynamic parametrization."""
+
+    # default parameterizations
+    default_warmup_shape = [[(64, 20, 4)]]
+    default_max_num_seqs = [4]
+    default_max_model_len = [256]
+
+    existing_markers = [
+        marker.name if marker.name != "parametrize" else marker.args[0]
+        for marker in metafunc.definition.own_markers
+    ]
+
+    marker = metafunc.config.option.markexpr  # From CLI
+    if "full_model" in marker:
```

Comment:
Won't this also trigger in cases like `-m "not full_model"`? Is that okay because we never expect to negate the `full_model` marker?